controllers/k6_reconcile.go

```diff
@@ -0,0 +1,50 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/grafana/k6-operator/api/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Reconcile k6 status with job status
+func ReconcileJobs(ctx context.Context, log logr.Logger, k6 *v1alpha1.K6, r *K6Reconciler) (ctrl.Result, error) {
+	selector := labels.SelectorFromSet(map[string]string{
+		"app":   "k6",
+		"k6_cr": k6.Name,
+	})
+
+	opts := &client.ListOptions{LabelSelector: selector, Namespace: k6.Namespace}
+	jl := &batchv1.JobList{}
+
+	if err := r.List(ctx, jl, opts); err != nil {
+		log.Error(err, "Could not list jobs")
+		return ctrl.Result{}, err
+	}
+
+	//TODO: We should distinguish between suceeded/failed
```

Comment:
Yes, they definitely need to be distinguished. Apparently, right now it's possible to get this kind of behavior:
```
2021-12-07T15:42:54.230Z        INFO    controllers.K6  10/10 pods ready        {"k6": "default/k6-sample"}
2021-12-07T15:42:56.109Z        DEBUG   controller      Successfully Reconciled {"reconcilerGroup": "k6.io", "reconcilerKind": "K6", "controller": "k6", "name": "k6-sample", "namespace": "d
efault"}
2021-12-07T15:42:56.111Z        INFO    controllers.K6  1/11 jobs complete      {"k6": "default/k6-sample"}
2021-12-07T15:42:56.111Z        DEBUG   controller      Successfully Reconciled {"reconcilerGroup": "k6.io", "reconcilerKind": "K6", "controller": "k6", "name": "k6-sample", "namespace": "d
efault"}
2021-12-07T15:42:56.113Z        INFO    controllers.K6  0/11 jobs complete      {"k6": "default/k6-sample"}
2021-12-07T15:42:56.210Z        DEBUG   controller      Successfully Reconciled {"reconcilerGroup": "k6.io", "reconcilerKind": "K6", "controller": "k6", "name": "k6-sample", "namespace": "d
efault"}
2021-12-07T15:43:06.417Z        INFO    controllers.K6  0/11 jobs complete      {"k6": "default/k6-sample"}
2021-12-07T15:43:06.417Z        DEBUG   controller      Successfully Reconciled {"reconcilerGroup": "k6.io", "reconcilerKind": "K6", "controller": "k6", "name": "k6-sample", "namespace": "d
efault"}
2021-12-07T15:43:08.506Z        INFO    controllers.K6  1/11 jobs complete      {"k6": "default/k6-sample"}
2021-12-07T15:43:08.506Z        DEBUG   controller      Successfully Reconciled {"reconcilerGroup": "k6.io", "reconcilerKind": "K6", "controller": "k6", "name": "k6-sample", "namespace": "d
efault"}
2021-12-07T15:43:42.151Z        INFO    controllers.K6  2/11 jobs complete      {"k6": "default/k6-sample"}
2021-12-07T15:43:42.151Z        DEBUG   controller      Successfully Reconciled {"reconcilerGroup": "k6.io", "reconcilerKind": "K6", "controller": "k6", "name": "k6-sample", "namespace": "d
efault"}
...
```
So it's 1 -> 0 -> 1 -> 2 -> ...11. This can be quite confusing :slightly_smiling_face: This log happened in case with failure of a starter + launching a second starter but there might be other such cases. So being more verbose here would be best.
This is not a request change per se, just a comment: I think this improvement can be added at a later point too.